h2/src/test/org/h2/test/db/TestCases.java

```diff
@@ -278,6 +279,68 @@ private void testDependencies() throws SQLException {
         conn.close();
     }
 
+    private void testDropTable() throws SQLException {
+        final boolean prevVal = config.stdDropTableRestrict;
+        try {
+            for (final boolean newVal : new boolean[] { true, false }) {
+                config.stdDropTableRestrict = newVal;
+                _testDropTable();
+            }
+        } finally {
+            config.stdDropTableRestrict = prevVal;
+        }
+    }
+
+    private static enum DropDependent {
+        NONE, VIEW, FOREIGN_KEY;
+    }
```

Comment:
this is a little hard to follow - especially for test cases, I'd prefer some small about of repetition over harder to read code - rather just duplicate what you need than use this enum